lib/change_stream.js

```diff
@@ -460,6 +458,7 @@ function applyKnownOptions(target, source, optionNames) {
 
 // This method performs a basic server selection loop, satisfying the requirements of
 // ChangeStream resumability until the new SDAM layer can be used.
+// TODO: use new SDAM layer
```

Comment:
Added to increase visibility of above comment.

Reply:
I think you need to rebase these changes to 3.5/3.6. In `master` this comment is no longer relevant, and we should remove `waitForTopologyConnected` completely.

Reply:
> we should remove `waitForTopologyConnected` completely

That sounds much more involved than what I'm attempting to do here. My plan was to do it first on master and the backport it 3.5/3.6; if we want to do it differently in master, I'll rebase.